src/main/java/Map.java

```diff
@@ -1,32 +1,21 @@
+import org.json.JSONObject;
+
+import java.io.InputStream;
+import java.util.*;
+
 public class Map {
 
     private Area[][] map;
     private Coordinate entryPoint;
     private int mapSize;
 
     Map () {
-        mapSize = 0;
-        entryPoint = new Coordinate(0 ,0); // maybe read from file? JSON?
```

Comment:
This comment is confusing to me as someone reading your code. A bit of clarification here would be useful!